Mathlib/Algebra/Polynomial/Module/Basic.lean

```diff
@@ -156,12 +156,14 @@ theorem smul_single_apply (i : ℕ) (f : R[X]) (m : M) (n : ℕ) :
 
 theorem smul_apply (f : R[X]) (g : PolynomialModule R M) (n : ℕ) :
     (f • g) n = ∑ x ∈ Finset.antidiagonal n, f.coeff x.1 • g x.2 := by
-  induction' f using Polynomial.induction_on' with p q hp hq f_n f_a
-  · rw [add_smul, Finsupp.add_apply, hp, hq, ← Finset.sum_add_distrib]
+  induction f using Polynomial.induction_on' with
+  | h_add p q hp hq =>
```

Comment:
Let's make a PR to correct these names (separately)

Reply:
I think they're fine? What names do you propose for the cases?

Reply:
The convention is to drop the `h_`; consider `induction n`, where we write `| zero` and `| succ n ih` not `| h_zero` and `| h_succ`.

So "name the hypothesis (concisely) as if it was a lemma" applies to induction principles.

Reply:
#23358